src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -1542,9 +1577,11 @@ void Terminal::_updateUrlDetection()
     }
 }
 
+// NOTE: This is the version of AddMark that comes from the UI. The VT api call into this too.
 void Terminal::AddMark(const Microsoft::Console::VirtualTerminal::DispatchTypes::ScrollMark& mark,

```

Comment:
nit: The naming overlap makes it hard to tell which is happening;

```
AddMark(mark);
AddMark(mark, pos, pos, false);
```

One of those doesn't shout "UI" to me more than the other.